web/models/setting/settings.ex

```diff
@@ -0,0 +1,75 @@
+defmodule CoursePlanner.Settings do
+  @moduledoc """
+  This module provides custom functionality for controller over the model
+  """
+  use CoursePlanner.Web, :model
+
+  alias CoursePlanner.{Repo, SystemVariable}
+  alias Ecto.Multi
+
+  @all_query               from sv in SystemVariable, select: {sv.id, sv}, order_by: :key
+  @visible_settings_query  from sv in SystemVariable, where: sv.visible  == true, order_by: :key
+  @editable_settings_query from sv in SystemVariable, where: sv.editable == true, order_by: :key
+
+  def all do
+    Repo.all(@all_query)
+  end
+
+  def get_visible_systemvariables do
+    Repo.all(@visible_settings_query)
+  end
+
+  def get_editable_systemvariables do
+    Repo.all(@editable_settings_query)
+  end
+
+  def get_changesets_for_update(setting_params) do
+    system_variables = all() |> Enum.into(Map.new)
+    setting_params = Enum.sort_by(setting_params, &(elem(&1, 1)["key"]))
```

Comment:
Since we are iterating based on the setting that are coming from form it's not for sure that they will come in a specific order, so I have to sort them like I order the read so Show, edit and update render the system_variables in the same way.